streams/src/main/java/org/apache/kafka/streams/kstream/internals/StatelessProcessorNode.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.internals.InternalTopologyBuilder;
+
+/**
+ * Used to represent any type of stateless operation:
+ *
+ * map, mapValues, flatMap, flatMapValues, filter, filterNot, branch
+ *
+ */
+class StatelessProcessorNode<K, V> extends StreamGraphNode<K, V> {
+
+    private final ProcessorSupplier processorSupplier;
+
+
+    StatelessProcessorNode(final String predecessorNodeName,
```

Comment:
Could we just use `ProcessorParameters` for the second and third parameters here?

Reply:
ack

Reply:
I understand it is only for reducing the number of parameters, but even for graph nodes where one processor node is likely going to be generated, we can still use it to reduce num. parameters right?

Reply:
ack, I'll update to use across the board.